### DISCUSSION TITLE
Circle-Map Realign error

### DISCUSSION THREAD

**Discussion by ZHONGHONGXIN** [ShixiangWang/self-study#103] [Q&A] [answered]
您好，我今天在Geogle Git-hub拜读了您关于[使用Circle-Map Realign鉴定环形DNA](https://shixiangwang.github.io/blog/use-circle-map-for-detecting-eccdna)的帖子，按照您的代码成功处理了biopython版本问题，但是在第四步：检测环形DNA（Circle-Map Realign）时，出现以下报错，如果您有空的话，麻烦给我一下建议。
      代码如下：
      source software/conda/bin/activate circle-map
Circle-Map Realign -i sort_circular_read_candidates.bam -qbam qname_unknown_circle.bam -sbam sorted_unknown_circle.bam -fasta hg38.fa -o my_unknown_circle.bed
      报错信息如下：
***** ERROR: Requested column 4, but database file - only has fields 1 - 0.
[E::idx_find_and_load] Could not retrieve index file for '../bwa/qname_unknown_circle.bam'
software/conda/envs/circle-map/lib/python3.8/site-packages/numpy/core/from
numeric.py:3474: RuntimeWarning: Mean of empty slice.
  return _methods._mean(a, axis=axis, dtype=dtype,
software/conda/envs/circle-map/lib/python3.8/site-packages/numpy/core/_met
hods.py:189: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
conda/envs/circle-map/lib/python3.8/site-packages/numpy/core/_met
hods.py:264: RuntimeWarning: Degrees of freedom <= 0 for slice
  ret = _var(a, axis=axis, dtype=dtype, out=out, ddof=ddof,
software/conda/envs/circle-map/lib/python3.8/site-packages/numpy/core/_met
hods.py:222: RuntimeWarning: invalid value encountered in true_divide
  arrmean = um.true_divide(arrmean, div, out=arrmean, casting='unsafe',
software/conda/envs/circle-map/lib/python3.8/site-packages/numpy/core/_met
hods.py:256: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
^M  0%|          | 0/100 [00:00<?, ?it/s]
^M0it [00:00, ?it/s]ESC[A^M  1%|          | 1/100 [00:00<00:15,  6.27it/s]
^M1it [00:00,  6.51it/s]ESC[A^M  5%|▌         | 5/100 [00:00<00:04, 19.43it/s]
^M5it [00:00, 19.75it/s]ESC[A^M  8%|▊         | 8/100 [00:00<00:04, 20.62it/s]
^M8it [00:00, 20.83it/s]ESC[A^M 15%|█▌        | 15/100 [00:00<00:02, 36.28it/s]
^M15it [00:00, 36.53it/s]ESC[A^M 20%|██        | 20/100 [00:00<00:02, 39.94it/s]
^M20it [00:00, 40.12it/s]ESC[A^M 26%|██▌       | 26/100 [00:00<00:01, 45.53it/s]
^M26it [00:00, 45.68it/s]ESC[A^M 31%|███       | 31/100 [00:00<00:01, 46.01it/s]
^M31it [00:00, 46.12it/s]ESC[A^M 36%|███▌      | 36/100 [00:00<00:01, 45.14it/s]
^M36it [00:00, 45.20it/s]ESC[A^M 41%|████      | 41/100 [00:01<00:01, 42.77it/s]
^M41it [00:01, 42.82it/s]ESC[A^M 46%|████▌     | 46/100 [00:01<00:01, 40.12it/s]
^M46it [00:01, 40.15it/s]ESC[A^M 51%|█████     | 51/100 [00:01<00:01, 41.09it/s]
^M51it [00:01, 41.08it/s]ESC[A^M 56%|█████▌    | 56/100 [00:01<00:01, 41.16it/s]
^M56it [00:01, 41.18it/s]ESC[A^M 61%|██████    | 61/100 [00:01<00:00, 39.33it/s]
^M61it [00:01, 39.35it/s]ESC[A^M 65%|██████▌   | 65/100 [00:01<00:00, 37.60it/s]
^M65it [00:01, 37.62it/s]ESC[A^M 71%|███████   | 71/100 [00:01<00:00, 43.14it/s]
^M71it [00:01, 43.15it/s]ESC[A^M 76%|███████▌  | 76/100 [00:02<00:00, 37.20it/s]
^M76it [00:01, 37.19it/s]ESC[A^M 84%|████████▍ | 84/100 [00:02<00:00, 43.80it/s]
^M84it [00:02, 43.77it/s]ESC[A^M 91%|█████████ | 91/100 [00:02<00:00, 47.26it/s]
^M91it [00:02, 47.28it/s]ESC[A^M 97%|█████████▋| 97/100 [00:02<00:00, 42.93it/s]
^M97it [00:02, 42.85it/s]ESC[A^M100it [00:02, 41.12it/s]
^M100%|██████████| 100/100 [00:02<00:00, 41.01it/s]
Traceback (most recent call last):
 File "/public/home/xylu_gibh/software/conda/envs/circle-map/bin/Circle-Map", line 10, in <module>
    sys.exit(main())
  File "/public/home/xylu_gibh/software/conda/envs/circle-map/lib/python3.8/site-packages/circlemap/circle_map.py", line 1164, in main
    run = circle_map()
  File "/public/home/xylu_gibh/software/conda/envs/circle-map/lib/python3.8/site-packages/circlemap/circle_map.py", line 187, in __init__
    output = merge_final_output(self.args.sbam, self.args.output, begin, self.args.split,
  File "/public/home/xylu_gibh/software/conda/envs/circle-map/lib/python3.8/site-packages/circlemap/utils.py", line 1262, in merge_final_output
    second_merging_round = unparsed_pd.sort_values(['chrom', 'start', 'end']).reset_index()
  File "/public/home/xylu_gibh/software/conda/envs/circle-map/lib/python3.8/site-packages/pandas/util/_decorators.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "/public/home/xylu_gibh/software/conda/envs/circle-map/lib/python3.8/site-packages/pandas/core/frame.py", line 6304, in sort_values
    keys = [self._get_label_or_level_values(x, axis=axis) for x in by]
  File "/public/home/xylu_gibh/software/conda/envs/circle-map/lib/python3.8/site-packages/pandas/core/frame.py", line 6304, in <listcomp>
    keys = [self._get_label_or_level_values(x, axis=axis) for x in by]
  File "software/conda/envs/circle-map/lib/python3.8/site-packages/pandas/core/generic.py", line 1840, in _get_label_or_level_values
    raise KeyError(key)
KeyError: 'chrom'

谢谢！

**Comment by ZHONGHONGXIN** [comment, marked as answer]
您好，我找到决绝问题的方法了，谢谢！

**Reply by ShixiangWang** [reply]
Great to  see that.
